platform/ios/src/MGLMapViewDelegate.h

```diff
@@ -273,6 +273,10 @@ NS_ASSUME_NONNULL_BEGIN
  Touch frameworks. On the other hand, static annotation images use less memory
  and draw more quickly than annotation views.
  
+ The user location annotation view can also be customized via this method. When
+ `annotation` is `MGLUserLocation`, return an instance of a custom
+ `MGLUserLocationAnnotationView` subclass.
```

Comment:
Returning an MGLUserLocationAnnotationView should be acceptable too, thanks to https://github.com/mapbox/mapbox-gl-native/pull/6163/files#diff-20502b6bc02986a4398d0da84b2d4b94R3827.